pkg/efscreate/create.go

```diff
@@ -0,0 +1,171 @@
+package efscreate
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+	"os"
+	"strings"
+)
+
+const (
+	operatorName          = "create-efs-volume"
+	infraGlobalName       = "cluster"
+	secretNamespace       = "kube-system"
+	secretName            = "aws-creds"
+	STORAGECLASS_LOCATION = "STORAGEClASS_LOCATION"
+	MANIFEST_LOCATION     = "MANIFEST_LOCATION"
+)
+
+func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext) error {
+	// Create core clientset for core and infra objects
+	kubeClient := kubeclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
+	nodes, err := kubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("error listing nodes: %v", err)
+		return fmt.Errorf("error listing nodes: %v", err)
+	}
+
+	configClient := configclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
+	infra, err := configClient.ConfigV1().Infrastructures().Get(ctx, infraGlobalName, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("error listing infrastructures objects: %v", err)
+		return fmt.Errorf("error listing infrastructure objects: %v", err)
+	}
+	ec2Session, err := getEC2Client(ctx, kubeClient, infra.Status.PlatformStatus.AWS.Region)
+	if err != nil {
+		klog.Errorf("error getting aws client: %v", err)
+		return fmt.Errorf("error getting aws client: %v", err)
+	}
+
+	efs := NewEFS_Session(infra, ec2Session)
+
+	fsID, err := efs.CreateEFSVolume(nodes)
+	if err != nil {
+		klog.Errorf("error creating efs volume: %v", err)
+		return err
+	}
+	klog.Infof("created fsID: %s", fsID)
+	err = writeStorageClassFile(fsID)
+	if err != nil {
+		klog.Errorf("error writing storageclass to location %s: %v", os.Getenv(STORAGECLASS_LOCATION), err)
+		return err
+	}
+	err = writeCSIManifest("efs-cs")
+	if err != nil {
+		klog.Errorf("error writing manifest to location %s: %v", os.Getenv(MANIFEST_LOCATION), err)
+		return err
+	}
+
+	return nil
+}
+
+func writeStorageClassFile(fsID string) error {
+	fileName := os.Getenv(STORAGECLASS_LOCATION)
+	scContent := `
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: efs-sc
```

Comment:
Nit: since the name is replaced in the manifest below, a const + `${storageclassname}` would be more appropriate.